RequestKit/JSONPostRouter.swift

```diff
@@ -1,40 +1,46 @@
 import Foundation
 
 public protocol JSONPostRouter: Router {
-    func postJSON<T>(session: RequestKitURLSession, expectedResultType: T.Type, completion: (json: T?, error: ErrorType?) -> Void)
+    func postJSON<T>(session: RequestKitURLSession, expectedResultType: T.Type, completion: (json: T?, error: ErrorType?) -> Void) -> URLSessionDataTaskProtocol?
 }
 
 public extension JSONPostRouter {
-    public func postJSON<T>(session: RequestKitURLSession = NSURLSession.sharedSession(), expectedResultType: T.Type, completion: (json: T?, error: ErrorType?) -> Void) {
+    public func postJSON<T>(session: RequestKitURLSession = NSURLSession.sharedSession(), expectedResultType: T.Type, completion: (json: T?, error: ErrorType?) -> Void) -> URLSessionDataTaskProtocol? {
+        guard let request = request() else {
+            return nil
+        }
+
+        let data: NSData
         do {
-            let data = try NSJSONSerialization.dataWithJSONObject(params, options: NSJSONWritingOptions())
-            if let request = request() {
-                let task = session.uploadTaskWithRequest(request, fromData: data) { data, response, error in
-                    if let response = response as? NSHTTPURLResponse {
-                        if response.statusCode != 201 {
-                            let error = NSError(domain: errorDomain, code: response.statusCode, userInfo: nil)
-                            completion(json: nil, error: error)
-                            return
-                        }
-                    }
+            data = try NSJSONSerialization.dataWithJSONObject(params, options: NSJSONWritingOptions())
+        } catch {
+            completion(json: nil, error: error)
+            return nil
+        }
 
-                    if let error = error {
+        let task = session.uploadTaskWithRequest(request, fromData: data) { data, response, error in
+            if let response = response as? NSHTTPURLResponse {
+                if response.statusCode != 201 {
```

Comment:
Handled in https://github.com/nerdishbynature/RequestKit/commit/bf2cc97b98f876e1564877186a0171054787f27c